x/evidence/types/querier.go

```diff
@@ -1,18 +1,25 @@
 package types
 
+import (
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+
+	query "github.com/cosmos/cosmos-sdk/types/query"
+)
+
 // Querier routes for the evidence module
 const (
 	QueryEvidence    = "evidence"
 	QueryAllEvidence = "all_evidence"
 )
 
-// QueryEvidenceParams defines the parameters necessary for querying Evidence.
-type QueryEvidenceParams struct {
-	EvidenceHash string `json:"evidence_hash" yaml:"evidence_hash"`
+// NewQueryEvidenceRequest creates a new instance of QueryEvidenceRequest.
+func NewQueryEvidenceRequest(hash tmbytes.HexBytes) *QueryEvidenceRequest {
+	return &QueryEvidenceRequest{EvidenceHash: hash}
 }
 
-func NewQueryEvidenceParams(hash string) QueryEvidenceParams {
-	return QueryEvidenceParams{EvidenceHash: hash}
+// NewQueryAllEvidenceRequest creates a new instance of QueryAllEvidenceRequest.
+func NewQueryAllEvidenceRequest(req *query.PageRequest) *QueryAllEvidenceRequest {
+	return &QueryAllEvidenceRequest{Req: req}
 }
 
 // QueryAllEvidenceParams defines the parameters necessary for querying for all Evidence.
```

Comment:
while we're at it, this also can be replaced with QueryAllEvidenceRequest

Reply:
I was thinking about doing this later as part of the CLI migration to use the new query service because `NewQueryAllEvidenceRequest` uses `*query.PageRequest` while `NewQueryAllEvidenceParams` still uses `page` and `limit` which are currently used as CLI command's flags. This would also affect the REST endpoint, see https://github.com/cosmos/cosmos-sdk/blob/marie/5921-grpc-x-evidence/x/evidence/client/rest/query.go#L67, I'm not sure if that's a concern at this point.

Reply:
In general we don’t want to change existing REST endpoints. So l would stick with the old parameter type for those if there is a difference